python/lsst/daf/butler/butler.py

```diff
@@ -200,23 +200,74 @@ def transaction(self):
             with self.datastore.transaction():
                 yield
 
+    def _standardizeArgs(self, datasetRefOrType, dataId=None, **kwds):
+        """Standardize the arguments passed to several Butler APIs.
+
+        Butler APIs that conceptually need a DatasetRef also allow passing a
```

Comment:
I'd make these two paragraphs a Notes section and move it to the bottom. A pattern I've noticed from our docstrings is a general abuse of the [Extended Summary](
https://developer.lsst.io/python/numpydoc.html#extended-summary) section. When I say the extended summary can be "A few sentences" I do mean it :) Having heavy summaries pushes the Parameters down the page by variable amounts and makes docs harder for readers to scan.